src/ostree/ot-admin-builtin-finalize-staged.c

```diff
@@ -51,10 +52,58 @@ ot_admin_builtin_finalize_staged (int argc, char **argv, OstreeCommandInvocation
   g_autoptr(GOptionContext) context = g_option_context_new ("");
   g_autoptr(OstreeSysroot) sysroot = NULL;
   if (!ostree_admin_option_context_parse (context, options, &argc, &argv,
-                                          OSTREE_ADMIN_BUILTIN_FLAG_SUPERUSER,
+                                          OSTREE_ADMIN_BUILTIN_FLAG_SUPERUSER | OSTREE_ADMIN_BUILTIN_FLAG_UNLOCKED,
                                           invocation, &sysroot, cancellable, error))
     return FALSE;
 
+  /* In case they're automounts, open /boot and the booted deployment on
+   * /sysroot so that the automounts don't expire. If they do expire and either
+   * gets unmounted, the service will be stopped.
+   */
+  int sysroot_fd = ostree_sysroot_get_fd (sysroot);
+  glnx_autofd int boot_fd = -1;
+  g_autoptr(OstreeDeployment) deployment = NULL;
+  g_autofree char *deployment_path = NULL;
```

Comment:
In new code I tend to prefer declare-and-initialize style (here on line 74).

Reply:
Fair. I ended up dropping the deployment handling here. I have noticed that in some scenarios gcc (at least on my system) seems to think you're using variables uninitialized when you don't explicitly initialize auto variables to `NULL`. I guess it probably is because of some code reordering. Anyways, that's tangential.